src/chat.rs

```diff
@@ -74,37 +208,38 @@ impl Chat {
                 Err(err.into())
             }
             Ok(mut chat) => {
-                match chat.id {
-                    DC_CHAT_ID_DEADDROP => {
-                        chat.name = context.stock_str(StockMessage::DeadDrop).into();
-                    }
-                    DC_CHAT_ID_ARCHIVED_LINK => {
-                        let tempname = context.stock_str(StockMessage::ArchivedChats);
-                        let cnt = dc_get_archived_cnt(context);
-                        chat.name = format!("{} ({})", tempname, cnt);
-                    }
-                    DC_CHAT_ID_STARRED => {
-                        chat.name = context.stock_str(StockMessage::StarredMsgs).into();
-                    }
-                    _ => {
-                        if chat.typ == Chattype::Single {
-                            let contacts = get_chat_contacts(context, chat.id);
-                            let mut chat_name = "Err [Name not found]".to_owned();
-
-                            if let Some(contact_id) = contacts.first() {
-                                if let Ok(contact) = Contact::get_by_id(context, *contact_id) {
-                                    chat_name = contact.get_display_name().to_owned();
-                                }
-                            }
+                if chat.id.is_deaddrop() {
```

Comment:
Maybe `ChatId` type should have been an enum. But it is an improvement anyway, let's merge this and then think about improvements.